### PR TITLE
Add ensure_pow method

### PR DESCRIPTION
**Pull request by lemunozm**
This PR extends #12967 with the `ensure_pow` function. I think this is also an interesting function to have along with the `ensure_ops` family methods.

`ensure_pow` acts similarly to `checked_pow` but returns an `ArithmeticError` instead.

Related #12754

### CODE REVIEW

File: primitives/runtime/src/traits.rs [hunk 1]
```diff
@@ -32,11 +32,11 @@ use impl_trait_for_tuples::impl_for_tuples;
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use sp_application_crypto::AppKey;
 pub use sp_arithmetic::traits::{
-	AtLeast32Bit, AtLeast32BitUnsigned, Bounded, CheckedAdd, CheckedDiv, CheckedMul, CheckedShl,
-	CheckedShr, CheckedSub, Ensure, EnsureAdd, EnsureAddAssign, EnsureDiv, EnsureDivAssign,
-	EnsureFixedPointNumber, EnsureFrom, EnsureInto, EnsureMul, EnsureMulAssign, EnsureOp,
-	EnsureOpAssign, EnsureSub, EnsureSubAssign, IntegerSquareRoot, One, SaturatedConversion,
-	Saturating, UniqueSaturatedFrom, UniqueSaturatedInto, Zero,
+	checked_pow, ensure_pow, AtLeast32Bit, AtLeast32BitUnsigned, Bounded, CheckedAdd, CheckedDiv,
```
Comment (lemunozm) [inline primitives/runtime/src/traits.rs:35]: I've reexported `checked_pow`. Please, double-check if you agree with it.

**Review by bkchr — APPROVED** [review]

File: primitives/runtime/src/traits.rs (hunk 1 above)
Comment (ggwpez) [reply primitives/runtime/src/traits.rs:35]: I think we will get rid of these anyway in https://github.com/paritytech/substrate/pull/12509

**Review by ggwpez — APPROVED** [review]
Please add some minimal success tests or merge in https://github.com/paritytech/substrate/pull/13043 and use the test structure from there.

**Comment by lemunozm** [comment]
I've implemented `ensure_pow` as a function because `checked_pow` is also implemented as a function. But maybe can be interesting to follow the same pattern as other `ensure_ops` and make a trait `EnsurePow` with an `ensure_pow` method. What do you think? I'm not sure why in `num-traits` the `checked_pow` is not implemented as a trait TBH.

File: primitives/runtime/src/traits.rs (hunk 1 above)
Comment (lemunozm) [reply primitives/runtime/src/traits.rs:35]: I've read the issue, could be a problem with our last move of `ArithmeticError` to `sp_arithmetic`?

**Comment by ggwpez** [comment]
> I've implemented `ensure_pow` as a function because `checked_pow` is also implemented as a function. But maybe can be interesting to follow the same pattern as other `ensure_ops` and make a trait `EnsurePow` with an `ensure_pow` method. What do you think? I'm not sure why in `num-traits` the `checked_pow` is not implemented as a trait TBH.

Dont know what is better here. Maybe there can be types that are not `CheckedMul` but can still have an `ensure_pow`?  
I cant imagine but that would not work with a free-standing function otherwise.

File: primitives/runtime/src/traits.rs (hunk 1 above)
Comment (ggwpez) [reply primitives/runtime/src/traits.rs:35]: No idea, I think its stale.

**Comment by ggwpez** [comment]
bot merge

**Comment by lemunozm** [comment]
> (Looks like the CI does not notice when you dismiss reviews 🤔)

Yes, it was weird, the review button switched between you and your partner, instead of selecting both 🤷‍♂️. 

Anyways, thanks for your reviews! 🙌🏻

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-37/1736/1

**Pull request merged**